post_request_task/task.py

```diff
@@ -29,48 +31,52 @@ def _start_queuing_tasks(**kwargs):
     If not called, tasks are delayed normally (so tasks still function without
     having to call _send_tasks_and_stop_queuing() manually when we're outside
     the request-response cycle.."""
-    _locals.__dict__['queue_tasks'] = True
+    _locals.queue_tasks = True
 
 
 def _stop_queuing_tasks(**kwargs):
     """Stops queuing tasks for this thread.
 
     Not supposed to be called directly, only useful for tests cleanup."""
-    _locals.__dict__.pop('queue_tasks', None)
+    _locals.queue_tasks = False
 
 
 def is_task_queuing_enabled_for_this_thread():
     """Returns whether post request task queuing is enabled for this thread."""
-    return _locals.__dict__.get('queue_tasks', False)
+    return _locals.queue_tasks
 
 
 def _send_tasks_and_stop_queuing(**kwargs):
     """Sends all delayed Celery tasks and stop queuing new ones for now."""
-    _locals.__dict__.pop('queue_tasks', None)
+    log.info('Stopping queueing tasks and sending already queued ones.')
+    _stop_queuing_tasks()
     queue = _get_task_queue()
     while queue:
-        cls, args, kwargs = queue.pop(0)
-        cls.original_apply_async(*args, **kwargs)
+        task, args, kwargs, extrakw = queue.pop(0)
+        task.original_apply_async(args=args, kwargs=kwargs, **extrakw)
```

Comment:
I looked at the [celery definition](https://github.com/celery/celery/blob/abf18635d714754d4057bc65f1950c8b145c7fc2/celery/app/task.py#L378) and yeah, this looks good.